lmfit/parameter.py

```diff
@@ -458,52 +458,34 @@ def set(self, value=None, vary=None, min=None, max=None, expr=None):
             self._val = value
         if vary is not None:
             self.vary = vary
-        if min is not None:
-            self.min = min
-        if max is not None:
-            self.max = max
+        if min is None:
+            min = -inf
+        if max is None:
+            max = inf
+        self.min = min
+        self.max = max
 
     def _init_bounds(self):
         """make sure initial bounds are self-consistent"""
         # _val is None means - infinity.
+        if self.max is None:
+            self.max = inf
+        if self.min is None:
+            self.min = -inf
         if self._val is not None:
-            if self.max is not None and self._val > self.max:
+            if self.min > self.max:
+                self._min, self._max = self.max, self.min
+            if isclose(self.min, self.max, atol=1e-13, rtol=1e-13):
```

Comment:
This `isclose` check needs to happen whenever `self.min` or `self.max` is changed. The only way to ensure that is to use properties! I suspect the majority of changing bounds does not go through the `set` method, they are changed directly, i.e. `par.min = 5`.
Otherwise the `self.from_internal` and `self.scale_gradient` can have divide by zero's, e.g. L571.